src/components/KTableData/KTableData.vue

```diff
@@ -172,91 +170,90 @@ import type {
+  emptyStateButtonAppearance,
+  emptyStateActionRoute,
+  error,
+  errorStateTitle = 'An error occurred',
+  errorStateMessage = 'Data cannot be displayed due to an error.',
+  errorStateActionMessage = '',
+  errorStateActionRoute,
+  maxHeight = 'none',
+  hidePagination,
+  paginationAttributes = {},
+  rowExpandable = () => false,
+  rowExpanded = () => false,
+  hideHeaders,
+  nested,
+  hidePaginationWhenOptional,
+  hideToolbar,
+  headers = [],
+  fetcherCacheKey = '',
+  cacheIdentifier = '',
+  searchInput = '',
+  initialFetcherParams = {},
+  clientSort,
+  sortable = true,
+  fetcher,
+  sortHandlerFunction,
+  tooltipTarget,
+  ...restProps
+} = defineProps<TableDataProps<H, D, O>>()
+
+const emit = defineEmits<TableDataEmits<H, D>>()
+
+const slots = defineSlots<TableDataSlots<H, D>>()
 
 const { useDebounce, useRequest, useSwrvState, clientSideSorter: defaultClientSideSorter } = useUtilities()
 
-const emit = defineEmits<{
-  (e: 'cell-click', value: { data: Record<string, any> }): void
-  (e: 'row-click', value: { data: Record<string, any> }): void
-  (e: 'error-action-click'): void
-  (e: 'empty-state-action-click'): void
-  (e: 'update:table-preferences', preferences: TablePreferences): void
-  (e: 'sort', value: TableSortPayload): void
-  (e: 'state', value: TableStatePayload): void
-  (e: 'row-select', data: Record<string, any>[]): void
-  (e: 'update:row-expanded', data: RowExpandPayload): void
-  (e: 'row-actions-toggle', data: RowActionsTogglePayload): void
-}>()
-
 const tableId = useId()
 
-const tableData = ref<Record<string, any>[]>([])
-const tableHeaders = computed((): TableDataHeader[] => props.sortable ? props.headers : props.headers.map((header) => ({ ...header, sortable: false })))
+const tableData = ref([]) as Ref<D[number][]>
+const tableHeaders = computed(() => sortable ? headers : headers.map((header) => ({ ...header, sortable: false })))
 const getEmptyStateButtonAppearance = computed((): ButtonAppearance => {
-  if (props.emptyStateButtonAppearance) {
-    return props.emptyStateButtonAppearance
+  if (emptyStateButtonAppearance) {
+    return emptyStateButtonAppearance
   }
 
-  return props.searchInput ? 'tertiary' : 'primary'
+  return searchInput ? 'tertiary' : 'primary'
 })
 
 const total = ref<number>(0)
 const page = ref<number>(1)
-const pageSize = ref<number>(getInitialPageSize(props.tablePreferences, props.paginationAttributes))
-const filterQuery = ref<string>(props.searchInput ?? '')
-const sortColumnKey = ref<string>('')
+const pageSize = ref<number>(getInitialPageSize(tablePreferences, paginationAttributes))
+const filterQuery = ref<string>(searchInput ?? '')
+const sortColumnKey = ref('') as Ref<ColumnKey>
 const sortColumnOrder = ref<SortColumnOrder>('desc')
-const offset: Ref<string | null> = ref(null)
-const offsets: Ref<Array<any>> = ref([])
+const offset = ref(null) as Ref<O | null>
+const offsets = ref([]) as Ref<(O | null)[]>
```

Comment:
The types here changed significantly - mind dropping the explanation for the change here?